python/lsst/pipe/base/pipeTools.py

```diff
@@ -153,14 +153,20 @@ def orderPipeline(pipeline: List[TaskDef]) -> List[TaskDef]:
     outputs = {}  # maps task index to its output DatasetType names
     allInputs = set()  # all inputs of all tasks
     allOutputs = set()  # all outputs of all tasks
+    dsTypeTaskLabels: dict[str, str] = {}  # maps DatasetType name to the label of its parent task
     for idx, taskDef in enumerate(pipeline):
         # task outputs
         dsMap = {name: getattr(taskDef.connections, name) for name in taskDef.connections.outputs}
         for dsTypeDescr in dsMap.values():
             if dsTypeDescr.name in allOutputs:
                 raise DuplicateOutputError(
-                    "DatasetType `{}' appears more than once as output".format(dsTypeDescr.name)
+                    "DatasetType `{}' in task `{}' already appears as an output in task `{}'.".format(
+                        dsTypeDescr.name,
+                        taskDef.label,
+                        dsTypeTaskLabels[dsTypeDescr.name],
```

Comment:
I think an f-string might be more readable here; it's a bit disorienting that the string is laid out horizontally while the substitution arguments are laid out vertically.